pkg/control/v2/client/wait/agent.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package wait
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+// ForAgent waits for the agent daemon to be able to be communicated with.
+func ForAgent(ctx context.Context, timeout time.Duration) error {
+	if timeout == 0 {
+		timeout = 1 * time.Minute // default of 1 minute
+	}
+	if timeout > 0 {
```

Comment:
I don't think we need this condition

Reply:
It is doing exactly that. If `-1` timeout is provided then it will run until the `ctx` is cancelled.

Reply:
Added it to `--help` output.